### PR TITLE
Integration test with docker-compose.

### PR DESCRIPTION
**Pull request by bpmcd**
New example using pytest fixtures.  This will start the SDC container once, and only once.

This changeset still needs actual verification of the results from SDC.

This is quite similar to the example @afoerster linked to in the previous pull request...

### CODE REVIEW

File: test/test_start_stop.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import subprocess
+from base_test_case import *
+
+def test_start(sdc):
+    assert 0 == subprocess.check_call(["../sdc-util", "pipeline", "start",
```
Comment (afoerster) [inline test/test_start_stop.py]: Should we use normal shell syntax here by default?
import shlex
...
cmd = shlex.split('../sdc-util pipeline stop ')
check_command(cmd)

File: test/base_test_case.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    return proc
+
+def get_docker_host():
+    docker_host = os.environ.get('DOCKER_HOST', '').strip()
+    if not docker_host:
+        return '127.0.0.1'
+    return docker_host
+
+def curl_until_success(port, endpoint="/", params={}):
+    for i in range(30):
```
Comment (afoerster) [inline test/base_test_case.py]: pull into constant?

File: test/base_test_case.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        try:
+            response = requests.get("http://{}:{}{}".format(
+                get_docker_host(), port, endpoint), params=params)
+        except requests.exceptions.ConnectionError:
+            pass
+        else:
+            return response
+        time.sleep(1)
```
Comment (afoerster) [inline test/base_test_case.py]: constant? I'm impatient, I'd make this shorter

File: test/base_test_case.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+import pytest
+import requests
+
+
+@pytest.fixture(scope="session")
+def sdc(request):
+    compose_file = "compose/sdc.yml"
+    popen_kwargs = {}
+    proc = subprocess.Popen(["docker-compose", "-f", compose_file, "up"], **popen_kwargs)
```
Comment (afoerster) [inline test/base_test_case.py]: Do you know where python gets the PATH from? Does this work when run in PyCharm?

File: test/test_start_stop.py (hunk 1 above)
Comment (afoerster) [inline test/test_start_stop.py]: could even pull out a little helper function that splits and checks

def check_cmd(cmd: String):
   .. split it
   .. assert

File: test/base_test_case.py (hunk 4 above)
Comment (bpmcd) [inline test/base_test_case.py]: I do not understand where the PATH comes from.  I'm assuming that it is inherited from the shell that created the process.
I don't use PyCharm, so I'm not sure...  I've been working mainly from the terminal with Vim.

File: test/base_test_case.py (hunk 2 above)
Comment (bpmcd) [inline test/base_test_case.py]: Will do.

File: test/base_test_case.py (hunk 3 above)
Comment (bpmcd) [inline test/base_test_case.py]: Will change to 0.25 like the old bash script.

File: test/test_start_stop.py (hunk 1 above)
Comment (bpmcd) [inline test/test_start_stop.py]: Sounds good.  I'll do some reading on shlex.

File: test/base_test_case.py (hunk 4 above)
Comment (afoerster) [inline test/base_test_case.py]: Looks like I'll just have to set the path manually when I'm using pycharm: https://stackoverflow.com/questions/8732119/pycharm-path-on-os-x

File: test/test_start_stop.py (hunk 1 above)
Comment (afoerster) [inline test/test_start_stop.py]: shlex will split the argumensts just like the bash cli argument parser will (honoring single and double quotes)

Comment (bpmcd) [inline test/test_start_stop.py]: Are you expecting the `assert` to validate the count of arguments? Or were you picturing something else?

**Comment by afoerster** [comment]
@bpmcd this all looks awesome my comments are nits, LGTM

**Comment by bpmcd** [comment]
OK. I've addressed the review comments and pushed the changes.  Holler if you see any other items that need alteration.

**Comment by bpmcd** [comment]
I think this is ready to merge!

**Pull request merged**